src/lib/storage/segment_encoding_utils.hpp

```diff
@@ -20,15 +20,27 @@ class BaseValueSegment;
 std::unique_ptr<BaseSegmentEncoder> create_encoder(EncodingType encoding_type);
 
 /**
- * @brief Encodes a segment by the given encoding specification (i.e., the encoding method
- * and -- when given and applicable -- the vector compression type)
+ * @brief Encodes a segment by the given encoding specification (i.e., the encoding method and -- when given and
+ * applicable -- the vector compression type). In general, do not use this method but rather
+ * ChunkEncoder::encode_segment() which supports reencoding to ValueSegments and avoids reencoding with
```

Comment:
avoids reencoding if the SegmentEncodingSpec is already in place?

Reply:
The comment sounds like `ChunkEncoder::encode_segment()` can do more and do it better than this method. What do we need it for?

Reply:
I left in place as most test use the "simple" method, which provides less convenience but always does exactly what is requested.
I don't see a good reason to keep it, but I wasn't sure we can remove the other function.

Remove it?

Reply:
You are the encoding master, I am not the one to judge on that.

Reply:
Changed it.